docs/features/software-catalog/life-of-an-entity.md

```diff
@@ -111,18 +111,55 @@ as possible.
 Each catalog deployment has a number of processors installed. They are
 responsible for receiving unprocessed entities that the catalog decided are due
 for processing, and then running that data through a number of processing
-stages. mutating the entity and emitting auxiliary data about it. When all of
+stages, mutating the entity and emitting auxiliary data about it. When all of
 that is done, the catalog takes all of that information and stores it as the
 processed entity, and errors and relations to other entities separately. Then,
 the catalog checks to see what entities are touched by that output, and triggers
 the final assembly of those (see Stitching below).
 
-There are several stages involved in the processing.
-
-> TODO: More info here
+![Processing overview](../../assets/features/catalog/life-of-an-entity_processing.svg)
+
+Entities are always processed one by one. Note how each processor can contribute
```

Comment:
I wonder if there's some better wording here, this makes it sound like it's not distributed and we don't parallelize. But I'm struggling to think of a better way to word it.

Reply:
aight that makes sense, tweaked wording